packages/request-client.js/src/api/request.ts

```diff
@@ -511,12 +470,15 @@ export default class Request {
   /**
    * Refresh the request data and balance from the network (check if new events happened - e.g: accept, payments etc..) and return these data
    *
+   * @param requestAndMeta return from getRequestFromId to avoid asking twice
    * @returns Refreshed request data
    */
-  public async refresh(): Promise<Types.IRequestData> {
-    const requestAndMeta: RequestLogicTypes.IReturnGetRequestFromId = await this.requestLogic.getRequestFromId(
-      this.requestId,
-    );
+  public async refresh(
+    requestAndMeta?: RequestLogicTypes.IReturnGetRequestFromId,
```

Comment:
I don't get this change

Reply:
In some case we have called `this.requestLogic.getRequestFromId();` just before calling this function.

It reduces the number of call to `this.requestLogic.getRequestFromId();`